evaluator/builtin_string.go

```diff
@@ -186,6 +187,26 @@ func builtinReverse(args []types.Datum, _ context.Context) (d types.Datum, err e
 	}
 }
 
+// See: http://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_space
+func builtinSpace(args []types.Datum, _ context.Context) (d types.Datum, err error) {
+	x := args[0]
+	switch x.Kind() {
+	case types.KindNull:
+		return d, nil
+	default:
+		v, err := x.ToInt64()
+		if err != nil || v < 0 {
+			v = 0
+		}
+		if v > math.MaxInt32 {
```

Comment:
It is more complicated, it depends on max_allowed_packet. We can keep it simple. Just set a threshold value.
  if ((ulonglong) count > INT_MAX32)
    count= INT_MAX32;

  // Safe length check
  tot_length= (uint) count * cs->mbminlen;
  if (tot_length > current_thd->variables.max_allowed_packet)
  {
    push_warning_printf(current_thd, Sql_condition::SL_WARNING,
                        ER_WARN_ALLOWED_PACKET_OVERFLOWED,
                        ER(ER_WARN_ALLOWED_PACKET_OVERFLOWED),
                        func_name(),
                        current_thd->variables.max_allowed_packet);
    goto err;
   }